src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java

```diff
@@ -473,4 +484,17 @@ private void saveTextBlocks(final Set<TextBlock> textBlocks, final TextSubmissio
             }
         }
     }
+
+    /**
+     * Send feedback to Athena (if enabled for both the Artemis instance and the exercise).
+     */
+    private void sendFeedbackToAthena(final TextExercise exercise, final TextSubmission textSubmission, final List<Feedback> feedbacks) {
+        if (athenaFeedbackSendingService.isPresent() && exercise.isFeedbackSuggestionsEnabled()) {
+            // Athena needs IDs on all feedbacks
+            if (feedbacks.stream().anyMatch(feedback -> feedback.getId() == null)) {
+                throw new IllegalArgumentException("All feedbacks need to have an ID");
```

Comment:
(Can this even happen? I assume not really since this is called directly after saving the feedback).

I don't like throwing an exception here, since this will make the request to save the feedback fail, even though the tutor feedback was saved and only an error related to Athena happened. Maybe just log a warning and don't seen the feedback in this case?

Reply:
I don't think this can ever happen, so I now removed this. If there are any problems related to this, we will notice them using the Athena logs.